DPL: report streaming state to the driver

ktf opened the pull request:
It will then be responsibility of the driver to decide what to do.
For the moment we simply quit on idle state.

The GUI has also been updated to display the state as color code for the
boxes.

shahor02 commented:
@ktf adding ``DeviceInfo::streamingState`` conflicts with existing initialization sequence in  https://github.com/AliceO2Group/AliceO2/blob/2775523642938bd286c60099a957e526c5c38938/Framework/Core/test/test_GUITests.cxx#L171
I had to move new datamember to the end of the DeviceInfo to compile it.

ktf commented:
We have fixed the compilation (on my branch that part is actually now in a separate DebugGUI repository, sorry about that) and changed the visualisation in the GUI to show the `StreamingState`. Errors are now marked as extra text in the title of the node.
